pkg/plugin/keyvault.go

```diff
@@ -69,7 +76,14 @@ func newKeyVaultClient(config *config.AzureConfig, vaultName, keyName, keyVersio
 		return nil, fmt.Errorf("failed to get vault url, error: %+v", err)
 	}
 
-	klog.InfoS("using kms key for encrypt/decrypt", "vaultName", vaultName, "keyName", keyName, "keyVersion", keyVersion)
+	klog.InfoS("using kms key for encrypt/decrypt", "vaultURL", *vaultURL, "keyName", keyName, "keyVersion", keyVersion)
+
+	if proxyMode {
+		proxyEndpoint := fmt.Sprintf("http://%s:%d/%s", proxyAddress, proxyPort, strings.TrimLeft(*vaultURL, "https://"))
+		klog.InfoS("proxy url", "url", proxyEndpoint)
+		vaultURL = &proxyEndpoint
```

Comment:
Maybe move this to a util function so we can unit test this?

Reply:
fixed